torch/_dynamo/variables/builder.py

```diff
@@ -1169,6 +1169,30 @@ def wrap_to_fake_tensor_and_record(
         assert source is not None
         static_shapes, reason = tensor_always_has_static_shape(e, is_tensor)
 
+        name = source.name()
+
+        # Prep for automatic dynamic
```

Comment:
Note - This algorithm could ostensibly be in VariableTracker creation for tensors, invoked through wrap_fx_proxy. This feels cleaner, but open to bikeshedding on where to put it.